File: ios/Khrysalis/com/lightningkite/khrysalis/location/GeoCoordinate.actual.swift
//Stub file made with Khrysalis 2 (by Lightning Kite)
import Foundation
import CoreLocation

//--- GeoCoordinate.distanceToMiles(GeoCoordinate)
public extension GeoCoordinate {
    func distanceToMiles(_ other: GeoCoordinate) -> Double {
        return CLLocation(latitude: self.latitude, longitude: self.longitude).distance(from: CLLocation(latitude: other.latitude, longitude: other.longitude)) * 0.000621371

    }
    func distanceToMiles(other: GeoCoordinate) -> Double {
        return distanceToMiles(other)
    }
}

File: android/src/main/java/com/lightningkite/khrysalis/location/GeoCoordinate.actual.kt
package com.lightningkite.khrysalis.location

import android.location.Location
import com.lightningkite.khrysalis.location.GeoCoordinate

fun GeoCoordinate.distanceToMiles(other: GeoCoordinate): Double {
    val loc1 = Location("")
    loc1.latitude = this.latitude
    loc1.longitude = this.longitude
    val loc2 = Location("")
    loc2.latitude = other.latitude
    loc2.longitude = other.longitude

    //meters to miles
    return loc1.distanceTo(loc2) / 1609.34
}
